fairscale/optim/adascale.py

```diff
@@ -453,17 +459,19 @@ def _final_callback(self) -> None:
         total_grad_sqr = np.array(
             [sum(param.grad.pow(2).sum().item() for param in group["params"]) for group in self._optimizer.param_groups]
         )
-        # Divide by (_num_grads_to_accum ** 2) to account for gradient
-        # accumulation.
-        if self._num_grads_to_accum > 1:
-            # np array doesn't support /=.
-            total_grad_sqr = total_grad_sqr / (self._num_grads_to_accum**2)
 
         # Wait for all_reduce to be done and move it to cpu & np.
         if work:
             work.wait()
         local_grad_sqr = self._local_grad_sqr.cpu().numpy()
 
+        if self._num_grads_to_accum > 1:
+            # Handle scaling for for gradient accumulation
+            if self._is_scaled_loss:
+                local_grad_sqr *= self._num_grads_to_accum**2
+            else:
+                total_grad_sqr /= self._num_grads_to_accum**2
```

Comment:
interesting, `/=` not worked in the past, maybe newer numpy works with it now?

Reply:
I think if you use `a /= 2` when `a` is an integer array it will break since the result will be float, otherwise it should be fine?